pgxn/neon_walredo/walredoproc.c

```diff
@@ -895,7 +895,12 @@ apply_error_callback(void *arg)
 	StringInfoData buf;
 
 	initStringInfo(&buf);
-	xlog_outdesc(&buf, record);
+#if PG_VERSION_NUM >= 150000
+	if (record->record)
+#else
+	if (record->decoded_record)
+#endif
+		xlog_outdesc(&buf, record);
 
 	/* translator: %s is a WAL record description */
 	errcontext("WAL redo at %X/%X for %s",
```

Comment:
will the `record->ReadRecPtr` be valid if `record->decoded_record` is not?

Reply:
It is LSN of the point from which we try to decode record. It should contain some meaningful content. But in any case - it is not real "pointer" which we can dereference, so even if it contains some garbage it can not cause SIGSEGV

Reply:
Agreed. The garbage could be side-tracking in some investigation, but let's focus on it later.